fix example data in docs

loru88 opened the pull request:
### Description
- some examples were based on legacy Elasticsearch version
- switch to string instead of number to match the OpenSearch type
- add deprecation message to opensearch_ism_policy_mapping

### Issues Resolved


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

loru88 commented:
partial duplicate of #165

Also it shows the same deprecation message in the documentation merged in #159

prudhvigodithi commented:
Thanks @loru88 LGTM, @rblcoder can you please add your review as well?

rblcoder commented:
@loru88 Few of the changes remain to be replicated in the tests as well.  They are minor ones. Can you please change the corresponding tests as well?

loru88 commented:
> @loru88 Few of the changes remain to be replicated in the tests as well. They are minor ones. Can you please change the corresponding tests as well?

Thank you, I just fixed the issues. I updated the `.go`. files and I also dixed the OpenSearch spelling because it was only fixed in the generated file but it got reverted every time I run the command `tfplugindocs`.

rblcoder reviewed (approved):
Thank you @loru88 , the updated examples are working.

loru88 commented:
that's great, thank you. Can you approve the Workflow run, please?
